src/code/z_map_exp.c

```diff
@@ -26,33 +33,38 @@ s32 Map_GetDungeonOrBossAreaIndex(PlayState* play) {
 }
 
 /**
- * Returns true if the current scene is a dungeon or boss scene, false otherwise.
+ * Tests if the current scene is in a dungeon or boss scene.
+ * @return true if the current scene is a dungeon or boss scene, false otherwise.
  */
-s32 Map_IsInDungeonOrBossArea(PlayState* play) {
-    if (Map_GetDungeonOrBossAreaIndex(play) == -1) {
+s32 Map_IsInDungeonOrBossScene(PlayState* play) {
+    if (MapExp_TryGetDungeonSceneIndex(play) == -1) {
         return false;
     }
 
     return true;
 }
 
-s32 func_8010A0A4(PlayState* play) {
-    if ((Map_GetDungeonOrBossAreaIndex(play) == -1) || !func_80102EF0(play)) {
+s32 MapExp_CurRoomHasMapI(PlayState* play) {
+    if ((MapExp_TryGetDungeonSceneIndex(play) == -1) || !MapDisp_CurRoomHasMapI(play)) {
         return false;
     }
 
     return true;
 }
 
-s32 sDungeonSceneIds[] = {
-    SCENE_MITURIN, SCENE_HAKUGIN, SCENE_SEA, SCENE_INISIE_N, SCENE_INISIE_R,
-};
-
 /**
- * If the current scene is a dungeon scene, this function returns an index
- * indicating which one it is. Otherwise, it returns -1.
+ * Tests if the current scene is a dungeon scene, excluding boss scenes.
+ * @return -1 if not a dungeon scene, else returns the DungeonSceneIndex of the dungeon scene
  */
-s32 Map_GetDungeonAreaIndex(PlayState* play) {
+s32 MapExp_TryGetDungeonSceneIndexForDungeon(PlayState* play) {
```

Comment:
```suggestion
s32 Map_GetDungeonSceneIndexDungeon(PlayState* play) {
```
Again not sure `Try` or `For` adds anything

Reply:
Try I guess I can get rid of. For makes it read better imo.